### DISCUSSION TITLE
Lesson 11 createCollectible does not seem to create (tokenCounter still 0)

### DISCUSSION THREAD

**Discussion by shonendumm** [smartcontractkit/full-blockchain-solidity-course-py#801] [Q&A] [answered]
I downloaded the code from Patrick's repo, and for the AdvancedCollectible contract, I tried running deploy_and_create.py --network rinkeby.

On etherscan, the deployed contract showed the createCollectible transaction was successful, but when I tried to read the tokenCounter (like what Patrick did in the video), the tokenCounter returned 0 instead of 1.

Did anyone encounter this issue too?

Here's an example of the deployed contract https://rinkeby.etherscan.io/address/0x60e25CA8015C9D58AB400539E20BE291D8D57741#readContract

**Comment by shonendumm** [comment, marked as answer]
Hours later, when I checked the contract transactions and tokenCounter again, it seems to be updated to 2 now.

I think it's because the Chainlink VRFCoordinator takes some time to call the callback function fulfillRandomness that will mint the NFT. Hence there's some delay between createCollectible and actually minting the token.

**Reply by cromewar** [reply]
You are absolutely right, FullfillRandomness is an asynchronous operation and it could take time, on mainnet is faster tough. 
Nice catch @shonendumm
